### ISSUE TITLE
The horovod runs on multiple nodes by tensorflow benchmark, and the Loss changes to Nan.

### ISSUE THREAD

**Issue by LiJianbo-AI** [horovod/horovod#1051]
**Environment:**
1. Framework: (TensorFlow,)
2. Framework version: TF v1.13
3. Horovod version:0.16.1
4. MPI version:OpenMPI 4.0
5. CUDA version:10.0
6. NCCL version:2.4.2
7. Python version: 3.7
8. OS and version: Ubuntu16.04

running Code
https://github.com/tensorflow/benchmarks/tree/cnn_tf_v1.13_compatible/scripts/tf_cnn_benchmarks

running script
 horovodrun --verbose -np 32-H 9.88.35.61:8,9.88.35.62:8,9.88.35.63:8,9.88.35.64:8  python scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py --data_format=NCHW --data_name=imagenet --batch_size=256 --model=resnet50_v1.5 --data_dir=/mnt/nfs-104/imagenet --optimizer=momentum --variable_update=horovod --use_fp16=True --num_epochs=64 --datasets_num_private_threads=8 --display_every=100 --print_training_accuracy=True

issue：
The total_loss change to nan
[1,3]<stdout>:Step[1,3]<stdout>:	Img/sec	total_loss	top_1_accuracy	top_5_accuracy
[1,15]<stdout>:1	images/sec: 492.6 +/- 0.0 (jitter = 0.0)	9.844	0.004	0.008
[1,13]<stdout>:1	images/sec: 492.0 +/- 0.0 (jitter = 0.0)	10.155	0.000	0.004
[1,12]<stdout>:1	images/sec: 494.2 +/- 0.0 (jitter = 0.0)	10.427	0.000	0.004
[1,14]<stdout>:1	images/sec: 492.8 +/- 0.0 (jitter = 0.0)	13.988	0.000	0.004
[1,11]<stdout>:1	images/sec: 493.3 +/- 0.0 (jitter = 0.0)	14.295	0.000	0.004
[1,9]<stdout>:1	images/sec: 492.7 +/- 0.0 (jitter = 0.0)	9.269	0.008	0.016
[1,10]<stdout>:1	images/sec: 491.8 +/- 0.0 (jitter = 0.0)	13.514	0.000	0.000
[1,6]<stdout>:1	images/sec: 491.3 +/- 0.0 (jitter = 0.0)	9.754	0.004	0.008
[1,2]<stdout>:1	images/sec: 491.1 +/- 0.0 (jitter = 0.0)	12.324	0.000	0.012
[1,7]<stdout>:1	images/sec: 490.3 +/- 0.0 (jitter = 0.0)	10.848	0.000	0.004
[1,8]<stdout>:1	images/sec: 491.5 +/- 0.0 (jitter = 0.0)	10.547	0.000	0.004
[1,31]<stdout>:1	images/sec: 486.5 +/- 0.0 (jitter = 0.0)	14.930	0.012	0.020
[1,0]<stdout>:1	images/sec: 491.1 +/- 0.0 (jitter = 0.0)	11.152	0.000	0.000
[1,1]<stdout>:1[1,1]<stdout>:	[1,1]<stdout>:images/sec: 490.5 +/- 0.0 (jitter = 0.0)[1,1]<stdout>:	[1,1]<stdout>:11.041[1,1]<stdout>:	[1,1]<stdout>:0.000[1,1]<stdout>:	[1,1]<stdout>:0.000
[1,4]<stdout>:1	images/sec: 490.2 +/- 0.0 (jitter = 0.0)	10.922	0.000	0.004
[1,30]<stdout>:1	images/sec: 484.9 +/- 0.0 (jitter = 0.0)	11.952	0.000	0.008
[1,29]<stdout>:1	images/sec: 484.3 +/- 0.0 (jitter = 0.0)	10.375	0.004	0.004
[1,5]<stdout>:1	images/sec: 488.4 +/- 0.0 (jitter = 0.0)	10.335	0.000	0.000
[1,22]<stdout>:1	images/sec: 488.2 +/- 0.0 (jitter = 0.0)	10.650	0.000	0.004
[1,21]<stdout>:1	images/sec: 486.8 +/- 0.0 (jitter = 0.0)	12.658	0.000	0.000
[1,20]<stdout>:1	images/sec: 487.1 +/- 0.0 (jitter = 0.0)	11.963	0.004	0.004
[1,28]<stdout>:1	images/sec: 488.3 +/- 0.0 (jitter = 0.0)	63.568	0.004	0.004
[1,24]<stdout>:1	images/sec: 488.6 +/- 0.0 (jitter = 0.0)	12.401	0.000	0.004
[1,26]<stdout>:1	images/sec: 485.3 +/- 0.0 (jitter = 0.0)	9.653	0.000	0.008
[1,18]<stdout>:1	images/sec: 486.3 +/- 0.0 (jitter = 0.0)	10.068	0.000	0.008
[1,27]<stdout>:1	images/sec: 485.5 +/- 0.0 (jitter = 0.0)	10.616	0.000	0.004
[1,17]<stdout>:1	images/sec: 486.3 +/- 0.0 (jitter = 0.0)	12.357	0.000	0.008
[1,19]<stdout>:1	images/sec: 488.3 +/- 0.0 (jitter = 0.0)	9.866	0.004	0.008
[1,16]<stdout>:1	images/sec: 486.4 +/- 0.0 (jitter = 0.0)	11.410	0.000	0.004
[1,23]<stdout>:1	images/sec: 486.6 +/- 0.0 (jitter = 0.0)	10.015	0.000	0.000
[1,25]<stdout>:1	images/sec: 484.0 +/- 0.0 (jitter = 0.0)	10.590	0.000	0.000
[1,3]<stdout>:1	images/sec: 480.8 +/- 0.0 (jitter = 0.0)	9.649	0.000	0.000
[1,20]<stdout>:100	images/sec: 516.0 +/- 5.4 (jitter = 62.2)	nan	0.000	0.000
[1,19]<stdout>:100	images/sec: 516.0 +/- 5.5 (jitter = 62.0)	nan	0.000	0.004
[1,17]<stdout>:100	images/sec: 516.0 +/- 5.4 (jitter = 62.6)	nan	0.008	0.012
[1,21]<stdout>:100	images/sec: 516.0 +/- 5.4 (jitter = 62.6)	nan	0.000	0.008
[1,18]<stdout>:100	images/sec: 516.0 +/- 5.4 (jitter = 62.8)	nan	0.000	0.004
[1,16]<stdout>:100	images/sec: 516.0 +/- 5.5 (jitter = 63.7)	nan	0.004	0.008
[1,15]<stdout>:100	images/sec: 516.0 +/- 5.4 (jitter = 64.9)	nan	0.000	0.000
[1,23]<stdout>:100	images/sec: 516.0 +/- 5.4 (jitter = 60.7)	nan	0.000	0.000
[1,13]<stdout>:100	images/sec: 516.0 +/- 5.5 (jitter = 64.9)	nan	0.000	0.000
[1,22]<stdout>:100	images/sec: 516.0 +/- 5.4 (jitter = 62.9)	nan	0.000	0.004
[1,12]<stdout>:100	images/sec: 516.0 +/- 5.4 (jitter = 65.9)	nan	0.000	0.000
[1,11]<stdout>:100	images/sec: 516.0 +/- 5.4 (jitter = 64.3)	nan	0.000	0.008
[1,10]<stdout>:100	images/sec: 516.0 +/- 5.4 (jitter = 64.6)	nan	0.000	0.004
[1,9]<stdout>:100	images/sec: 516.0 +/- 5.5 (jitter = 65.5)	nan	0.000	0.000
[1,7]<stdout>:100	images/sec: 516.0 +/- 5.4 (jitter = 64.0)	nan	0.000	0.008
[1,5]<stdout>:100	images/sec: 516.0 +/- 5.4 (jitter = 64.4)	nan	0.000	0.012

**Comment by alsrgv** [comment]
@LiJianbo-AI, can you try reducing the learning rate?  You total batch size is quite large, too - 8192.

**Comment by fabboe** [comment]
@LiJianbo-AI do you have new insights on this? Experiencing the same on single machine run, with a training task that works well without horovod in the same config (batch size etc)

**Comment by karakusc** [comment]
@fabboe Make sure the effective learning rate used is the same when Horovod is used. Note that this script uses `hvd.allreduce` with the option `average=False`, which effectively scales the learning rate by the number of gpus. Make sure the learning rate is not scaled somewhere else too.

**Comment by fabboe** [comment]
@karakusc great tipp. Indeed there was a bug in my learning rate setting. Thanks!
